src/modules/socketClient/SocketClient.ts

```diff
@@ -0,0 +1,42 @@
+import { io, Socket } from "socket.io-client";
+
+class SocketClient {
+
+  socket: Socket
+  isSocketConnected: boolean
+  
+  constructor() {
+    this.isSocketConnected = false;
+    this.socket = io("http://localhost:5000");
+    
+    this.checkSocketConnection().then(() => {
+      this.isSocketConnected = this.socket.connected
+    })
+  }
+
+  async checkSocketConnection(): Promise<any>{
+    await new Promise((resolve, reject) => {
+      this.socket.on('connect', () => {
+        console.log('Socket Connected? ', this.socket.connected);
+        resolve(this.socket.connected)
+      });
+    })
+  }
+
+  getSocket() : Socket {
+    return this.socket;
+  }
```

Comment:
We don't want to expose the socket object directly

Reply:
Can you suggest a way to improve this?
We will need the socket object available throughout the application.

Reply:
Expose `sendMessage` and `getMessage` wrapper methods